homeassistant/helpers/config_validation.py

```diff
@@ -1769,10 +1771,32 @@ def STATE_CONDITION_SCHEMA(value: Any) -> dict[str, Any]:
     )
 )
 
+
+def _backward_compat_trigger_schema(value: Any | None) -> Any:
+    """Backward compatibility for trigger schemas."""
+
+    if not isinstance(value, dict):
+        return value
+
+    response = copy(value)
+
+    # `platform` has been renamed to `trigger`
+    # But we still use `platform` to maintain backward compatibility
+    if CONF_TRIGGER in value:
+        response[CONF_PLATFORM] = response.pop(CONF_TRIGGER)
+
+    return response
+
+
 TRIGGER_BASE_SCHEMA = vol.Schema(
     {
         vol.Optional(CONF_ALIAS): str,
-        vol.Required(CONF_PLATFORM): str,
+        vol.Exclusive(CONF_PLATFORM, "trigger"): str,
+        vol.Exclusive(
+            CONF_TRIGGER,
+            "trigger",
+            msg="Cannot specify both 'platform' and 'trigger'. Please use 'trigger' only.",
+        ): str,
```

Comment:
Revert these changes, these checks should be in the `_backward_compat_trigger_schema` if that runs first.